### PR TITLE
Event aggregates preview

### PR DESCRIPTION
**Pull request by muradsater**

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -161,6 +166,8 @@ export interface EventQuery {
   activeAtTimeRange: boolean;
   columns: string[];
   advancedFilter: string;
+  withAggregate: boolean;
+  property: string[];
```
Comment (polomani) [inline src/types.ts:170]: I suggest having this as a nested object:
```suggestion
  aggregate?: {
         type: "uniqueValues" | "count",
         properties: { property: string }[]
  }
```

File: src/datasources/EventsDatasource.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        return {
+          advancedFilter: advancedFilterQuery,
+          filter: { ...timeRange, ...params },
+          limit: EVENTS_PAGE_LIMIT,
+        };
+      }
+      return {
+        filter: { ...timeRange, ...params },
+        limit: EVENTS_PAGE_LIMIT,
+      };
```
Comment (polomani) [inline src/datasources/EventsDatasource.ts:92]:
```
filter: { ...timeRange, ...params },
```
this line is common for all the cases, you need to move it out
```
const filter =   { ...timeRange, ...params }
```

Comment (polomani) [reply src/datasources/EventsDatasource.ts:92]: also, rename the function.
data is too generic, maybe `getRequestBody`?

**Review by polomani — COMMENTED** [review]
Refactor is good, but we need to improve the json structure, by encapsulating aggregate params into one object.

**Comment by polomani** [comment]
Screenshot 2022-09-15 at 12 52 50
Also I didn't manage to get this to work, @KissSpinobleZsolt

**Comment by muradsater** [comment]
Did you specify an advanced filter query? It's not supported for normal
queries since it's deprecated in the API

**Comment by muradsater** [comment]
Advanced query could be
```
{
    "equals": {
        "property": ["RULE_BROKEN"],
        "value": "Anomaly"
    }
}
```

File: src/types.ts [hunk 3]
```diff
@@ -166,8 +169,11 @@ export interface EventQuery {
   activeAtTimeRange: boolean;
   columns: string[];
   advancedFilter: string;
-  withAggregate: boolean;
-  property: string[];
+  aggregate: {
```
Comment (polomani) [inline src/types.ts:172]: this must be an optional type

File: src/__tests__/datasource.test.ts [hunk 4]
```diff
@@ -650,8 +650,11 @@ describe('Datasource Query', () => {
           activeAtTimeRange: false,
           columns: [''],
           advancedFilter: '',
-          withAggregate: false,
-          property: [],
+          aggregate: {
```
Comment (polomani) [inline src/__tests__/datasource.test.ts:653]: pretty sure this is not supposed to be here.
aggregate is `undefined` for all legacy dashboards

**Pull request merged**